### PR TITLE
Custom preonic layout optimised for programming

### PR DESCRIPTION
**Pull request by AlexDaigre**
[Layout Images](https://imgur.com/a/k95XQlP)

# Don Gato's programming Layout

The default Preonic layout is suited to general english typing but not very convent to program on. This is my custom layout to re-optimise the keyboard for programming tasks.
I have been programming with this keyboard layout at work and I found it much faster and easier to adjust to than the default layout.

- Removed keys on the RAISE, LOWER, and ADJUST layers that duplicated keys on the main QWERTY layout.
- Moved the control, alt, and gui keys to a more natural position.
- Added the Caps-lock key on RAISE and LOWER in its position on a standard keyboard.
- Moved {}[], -_, and \| (on RAISE and LOWER) to be close to to similar keys on the QWERTY layout to make them easier to remember. They are kept as close as possible to their locations on a standard layout so users won't have to ADJUST their muscle memory much.
    - (), {}, and [] are all on the 9 and 0 keys. This is where () are placed naturally, so it is easy to find all of your braces.
    - -_ are moved to the += key. Now you can remember that -, +, and = are all together. The keys are also right next to each-other on a standard layout, so it is a very small move.
    - \ and | have beeb moved to the /? key. Easy to remember as / and \ are together.
- Moved backspace to make room for +=.
- Added delete to the backspace key on RAISE / LOWER
- Removed backlight and added an ADJUST key. As the Massdrop kits don't come with backlighting default, I think this is more useful for most users.
- I have left the Colemak and Dvorak alternate layouts, but they won't map up as conveniently to the other layers.
- I have included the keyboard layouts so that those wishing to make modifications or new versions of this layout can quickly update their images.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
The only issue I see here, is that we generally prefer to have the keymap folder name match the github user account name.  This isn't a hard requirement, but heavily preferred.

Otherwise, looks great!

File: keyboards/preonic/keymaps/programmer-dongato/keymap.c [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+  _ADJUST
+};
+
+enum preonic_keycodes {
+  QWERTY = SAFE_RANGE,
+  COLEMAK,
+  DVORAK,
+  LOWER,
+  RAISE,
+  BACKLIT,
```
Comment (noroadsleft) [inline keyboards/preonic/keymaps/programmer-dongato/keymap.c]: You're not using this keycode...

```suggestion
```

File: keyboards/preonic/keymaps/programmer-dongato/keymap.c [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+            writePinLow(E6);
+            #endif
+          } else {
+            unregister_code(KC_RSFT);
+            #ifdef __AVR__
+            writePinHigh(E6);
+            #endif
+          }
+          return false;
+          break;
```
Comment (noroadsleft) [inline keyboards/preonic/keymaps/programmer-dongato/keymap.c]: ... or its logic.

```suggestion
```

File: keyboards/preonic/keymaps/programmer-dongato/keymap.c (hunk 1 above)
Comment (AlexDaigre) [reply keyboards/preonic/keymaps/programmer-dongato/keymap.c]: Thank you. I have re-added the BACKLIT function to the mapping, so it is available should it be needed.

Comment (drashna) [reply keyboards/preonic/keymaps/programmer-dongato/keymap.c]: Just a heads up, if this is for a rev3, it doesn't use backlighting, at all, and can/should be safely removed.

**Comment by AlexDaigre** [comment]
@noroadsleft Thank you for the feedback. I have updated the keymap name. Let me know if there are any other issues.

**Comment by noroadsleft** [comment]
@AlexDaigre Looks good. Sorry for the late response.

**Pull request merged**